explorer/src/test/java/com/scalar/client/tool/explorer/ExplorerTest.java

```diff
@@ -53,119 +54,83 @@
   @Mock private ClientConfig clientConfig;
   @Mock private ClientService clientService;
   @Mock private ContractExecutionResponse contractExecutionResponseOK;
-  @Mock private ContractExecutionResponse contractExecutionResponseCertificateNotFound;
-  @Mock private ContractExecutionResponse contractExecutionResponseContractNotFound;
-  @Mock private LedgerServiceResponse ledgerServiceResponseOK;
-  @Mock private LedgerServiceResponse ledgerServiceResponseCertificateNotFound;
-  @Mock private LedgerServiceResponse responseListContract;
-  @Mock private LedgerValidationResponse responseValidation;
+  @Mock private ContractExecutionResult contractExecutionResultOK;
+  @Mock private ClientException clientException;
+  @Mock private LedgerValidationResult resultValidation;
   private Explorer explorer;
+  private ClientException clientExceptionCertificateNotFound;
+  private ClientException clientExceptionContractNotFound;
 
   @Rule public ExpectedException thrown = ExpectedException.none();
 
   @Before
   public void setUp() {
     MockitoAnnotations.initMocks(this);
 
+    clientExceptionCertificateNotFound = new ClientException("Certificate not found", StatusCode.CERTIFICATE_NOT_FOUND);
+    clientExceptionContractNotFound = new ClientException("contract not found", StatusCode.CONTRACT_NOT_FOUND);
+
     explorer = new Explorer(clientService, clientConfig);
     when(clientConfig.getCertHolderId()).thenReturn(HOLDER_ID);
-
-    when(clientService.registerCertificate()).thenReturn(ledgerServiceResponseOK);
-    when(clientService.registerContract(
-            anyString(), anyString(), anyString(), eq(Optional.empty())))
-        .thenReturn(ledgerServiceResponseOK);
-
-    when(ledgerServiceResponseOK.getStatus()).thenReturn(StatusCode.OK.get());
-    when(ledgerServiceResponseCertificateNotFound.getStatus())
-        .thenReturn(StatusCode.CERTIFICATE_NOT_FOUND.get());
-    when(ledgerServiceResponseCertificateNotFound.getMessage()).thenReturn("certificate not found");
-
-    when(contractExecutionResponseOK.getStatus()).thenReturn(StatusCode.OK.get());
-    when(contractExecutionResponseCertificateNotFound.getStatus())
-        .thenReturn(StatusCode.CERTIFICATE_NOT_FOUND.get());
-    when(contractExecutionResponseCertificateNotFound.getMessage())
-        .thenReturn("certificate not found");
-    when(contractExecutionResponseContractNotFound.getStatus())
-        .thenReturn(StatusCode.CONTRACT_NOT_FOUND.get());
-    when(contractExecutionResponseContractNotFound.getMessage()).thenReturn("contract not found");
   }
 
   @Test
   public void get_CertificateAndContractNotRegistered_ShouldSucceed() {
     // Arrange
-    when(contractExecutionResponseOK.getResult()).thenReturn("{ \"id\": \"bar\" }");
+    Optional<JsonObject> jsonObject = Optional.of(Json.createObjectBuilder().add("status", StatusCode.OK.get()).build());
```

Comment:
```suggestion
    JsonObject expected = Json.createObjectBuilder().add("status", StatusCode.OK.get()).build();
```

Reply:
Thank you for review comment.
Code updated with above modification.